### PR TITLE
(WIP) BLO-1888-Adapt-StarknetKit-to-new-JSON-RPC-Wallet-API

### PR DESCRIPTION
**Pull request by dhruvkelawala**

### CODE REVIEW

File: src/connectors/connector.ts [hunk 1]
```diff
@@ -45,7 +44,7 @@ export abstract class Connector extends EventEmitter<ConnectorEvents> {
   /** Disconnect wallet. */
   abstract disconnect(): Promise<void>
   /** Get current account. */
-  abstract account(): Promise<AccountInterface>
+  abstract account(): Promise<string>
```
Comment (bluecco) [inline src/connectors/connector.ts:47]: is still compatible with starknet-react by changing this?

File: src/connectors/injected/index.ts [hunk 2]
```diff
@@ -159,19 +162,24 @@ export class InjectedConnector extends Connector {
       throw new ConnectorNotFoundError()
     }
 
-    if (!this._wallet?.isConnected) {
-      throw new UserNotConnectedError()
-    }
+    // if (!this._wallet?.isConnected) {
```
Comment (bluecco) [inline src/connectors/injected/index.ts]: can be removed instead of commented?

File: src/connectors/connector.ts [hunk 3]
```diff
@@ -44,8 +43,8 @@ export abstract class Connector extends EventEmitter<ConnectorEvents> {
   abstract connect(): Promise<ConnectorData>
   /** Disconnect wallet. */
   abstract disconnect(): Promise<void>
-  /** Get current account. */
-  abstract account(): Promise<AccountInterface>
+  /** Get current account silently. Return null if the account is not authorized */
+  abstract account(): Promise<string | null>
```
Comment (bluecco) [inline src/connectors/connector.ts:47]: is still compatible with starknet-react by changing this?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "starknetkit",
-  "version": "1.1.3",
+  "version": "2.0.0",
```
Comment (vladutjs) [inline package.json:3]: are we sure we want to jump to 2.0.0?

File: src/connectors/connector.ts (hunk 3 above)
Comment (dhruvkelawala) [reply src/connectors/connector.ts:47]: We will have to create a PR on starknet-react with new changes. starknet-react doesn't support the new interface. Should I add Francesco to this review?

File: package.json (hunk 4 above)
Comment (bluecco) [reply package.json:3]: I think it makes sense since they're breaking changes

Comment (dhruvkelawala) [reply package.json:3]: It's technically a breaking change to the StarknetWindowObject. So yes, I think

File: src/helpers/defaultConnectors.ts [hunk 5]
```diff
@@ -10,7 +10,6 @@ import { WebWalletConnector } from "../connectors/webwallet"
 export const defaultConnectors = ({
   argentMobileOptions,
   webWalletUrl,
-  provider,
```
Comment (vladutjs) [inline src/helpers/defaultConnectors.ts:13]: can we still inject a custom provider if you remove this?

Comment (dhruvkelawala) [reply src/helpers/defaultConnectors.ts:13]: We don't inject a provider anymore with the new SNIP. Providers are something the dapp devs use themselves from sn.js

File: src/connectors/connector.ts (hunk 3 above)
Comment (vladutjs) [reply src/connectors/connector.ts:47]: any news about this, how we want to handle?

**Review by vladutjs — APPROVED** [review]

File: src/connectors/connector.ts (hunk 3 above)
Comment (bluecco) [reply src/connectors/connector.ts:47]: Francesco will have time next week to check too

File: package.json (hunk 4 above)
Comment (gergold) [reply package.json:3]: Using `semantic-release`, we should not change the version manually but reflect the breaking change in a commit message that will trigger a major release bump. see https://semantic-release.gitbook.io/semantic-release/

**Review by vladutjs — APPROVED** [review]

**Pull request merged**